pkg/clockchecker/clockchecker.go

```diff
@@ -0,0 +1,83 @@
+package clockchecker
+
+import (
+	"time"
+
+	"github.com/aukilabs/go-tooling/pkg/errors"
+	"github.com/aukilabs/go-tooling/pkg/logs"
+	"github.com/beevik/ntp"
+)
+
+// Configurable parameters
+var (
+	ntpServer        = "pool.ntp.org"
+	initialDelay     = 3 * time.Second // First check after 3 seconds
+	secondCheckDelay = 1 * time.Minute // Second check after 1 minute
+	checkInterval    = 6 * time.Hour   // Subsequent checks every 6 hours
+
+	warningThreshold = 5 * time.Second  // Warning if out of sync by more than 5s
+	errorThreshold   = 10 * time.Second // Error if out of sync by more than 10s
+
+	// Function variable for testing
+	getNTPTime = func() (time.Time, error) {
+		return ntp.Time(ntpServer)
+	}
+)
+
+// CheckClockSync checks if the system clock is out of sync and logs at different levels.
+func CheckClockSync() error {
```

Comment:
This should be lowercase probably as we don't need to use it outside of the module.